opensourceleg/actuators/dephy.py

```diff
@@ -0,0 +1,1739 @@
+        This is necessary because the magnetic output encoders are nonlinear.
+        By making the map while the joint is unloaded, joint position calculated by motor position * gear ratio
+        should be the same as the true joint position.
+
+        Output from this function is a file containing a_i values parameterizing the map
+
+        Eqn: position = sum from i=0^5 (a_i*counts^i)
+
+        Author: Kevin Best
+                U-M Locolab | Neurobionics Lab
+                Gitub: tkevinbest, https://github.com/tkevinbest
+        """
+
+        if not self.is_homed:
+            LOGGER.warning(
+                msg=f"[{self.__repr__()}] Please home the {self.tag} joint before making the encoder map."
+            )
+            return
+
+        if os.path.exists(f"./{self.tag}_encoder_map.npy") and not overwrite:
+            LOGGER.info(
+                msg=f"[{self.__repr__()}] Encoder map exists. Skipping encoder map creation."
+            )
+            return
+
+        self.set_control_mode(mode=CONTROL_MODES.CURRENT)
+        self.set_current_gains()
+        time.sleep(0.1)
+        self.set_current_gains()
+
+        self.set_joint_torque(value=0.0)
+        time.sleep(0.1)
+        self.set_joint_torque(value=0.0)
+
+        _joint_encoder_array = []
+        _output_position_array = []
+
+        LOGGER.info(
+            msg=f"[{self.__repr__()}] Please manually move the {self.tag} joint numerous times through its full range of motion for 10 seconds. \n{input('Press any key when you are ready to start.')}"
+        )
+
+        _start_time: float = time.time()
+
+        try:
+            while time.time() - _start_time < 10:
+                LOGGER.info(
+                    msg=f"[{self.__repr__()}] Mapping the {self.tag} joint encoder: {10 - time.time() + _start_time} seconds left."
+                )
+                self.update()
+                _joint_encoder_array.append(self.joint_encoder_counts)
+                _output_position_array.append(self.output_position)
+                time.sleep(1 / self.frequency)
+
+        except KeyboardInterrupt:
+            LOGGER.warning(msg="Encoder map interrupted.")
+            return
+
+        LOGGER.info(
+            msg=f"[{self.__repr__()}] You may now stop moving the {self.tag} joint."
+        )
+
+        _power = np.arange(4.0)
+        _a_mat = np.array(_joint_encoder_array).reshape(-1, 1) ** _power
+        _beta = np.linalg.lstsq(_a_mat, _output_position_array, rcond=None)
+        _coeffs = _beta[0]
+
+        self.set_encoder_map(np.polynomial.polynomial.Polynomial(coef=_coeffs))
+
+        np.save(file=f"./{self.tag}_encoder_map.npy", arr=_coeffs)
+        LOGGER.info(
+            msg=f"[{self.__repr__()}] Encoder map saved to './{self.tag}_encoder_map.npy'."
+        )
+
+    def set_motor_torque(self, value: float) -> None:
+        """
+        Sets the motor torque in Nm.
```

Comment:
I think these kind of doc strings can be made more helpful. 
"""
Sets the motor torque in Nm. 
This is the torque at the motor rotor, before the gearbox. 
"""